dbms/src/Storages/DeltaMerge/SegmentReadTaskPool.cpp

```diff
@@ -95,4 +95,170 @@ SegmentReadTasks SegmentReadTask::trySplitReadTasks(const SegmentReadTasks & tas
     return result_tasks;
 }
 
+BlockInputStreamPtr SegmentReadTaskPool::getInputStream(uint64_t seg_id, SegmentReadTaskPtr & t)
```

Comment:
What is the relationship of `t` with `this->tasks`?

Reply:
This method looks very weird to me.

Reply:
`t` is poped from `this-tasks` in `SegmentReadTaskScheduler::scheduleMergedTask`:
First, `scheduleSegmentUnlock` choose a segment to read.
Third, call `SegmentReadTaskPool::getTask` to pop task and build a `MergedTask`.

`MergedTask` includes `SegmentReadTaskPool`s and `SegmentReadTask`s.
Read threads use `SegmentReadTaskPool` and `SegmentReadTask` to build a segment inputstream before first read.

Reply:
I think we should reconsider the code design. Maybe later PRs. Currently, it looks OK to me.